sql/tsql/TSqlParser.g4

```diff
@@ -4115,6 +4220,43 @@ xml_data_type_methods
     | modify_method
     ;
 
+// https://learn.microsoft.com/en-us/sql/t-sql/functions/date-bucket-transact-sql?view=sql-server-ver16
+dateparts_9
+    : (YEAR | YEAR_ABBR)
+    | (QUARTER | QUARTER_ABBR)
+    | (MONTH | MONTH_ABBR)
+    | (DAY | DAY_ABBR)
+    | (WEEK | WEEK_ABBR)
+    | (HOUR | HOUR_ABBR)
+    | (MINUTE | MINUTE_ABBR)
+    | (SECOND | SECOND_ABBR)
+    | (MILLISECOND | MILLISECOND_ABBR)
+    ;
+
+// https://learn.microsoft.com/en-us/sql/t-sql/functions/dateadd-transact-sql?view=sql-server-ver16
+dateparts_12
+    :
+    | dateparts_9
+    | (DAYOFYEAR | DAYOFYEAR_ABBR)
+    | (MICROSECOND | MICROSECOND_ABBR)
+    | (NANOSECOND | NANOSECOND_ABBR)
+    ;
```

Comment:
Is it ok that `dateparts_12` has the empty alternative? If so, plese remove the empty alternative here and insert `?` to all `dateparts_12` usings. Also, parentheses are excess.

```suggestion
dateparts_12
    :
    | dateparts_9
    | DAYOFYEAR | DAYOFYEAR_ABBR
    | MICROSECOND | MICROSECOND_ABBR
    | NANOSECOND | NANOSECOND_ABBR
    ;
```

Reply:
Oh, empty alternative isn't okey here. This is my inattention. Parentheses also deleted.